File: CodeStudio/binary-strings-with-no-consecutive-1s/main.cpp
void solve(string pref, int n, vector<string>& ans) {

    if(pref.length() == n) {
        ans.emplace_back(pref);
        return;
    }

    string temp;

    if(pref == "" || pref[pref.length()-1] == '0') {
        // Set the state 
        temp = pref + '0';

        // Goto next state
        solve(temp, n, ans);

        // Set the state
        temp = pref + '1';

        // Goto next state
        solve(temp, n, ans);

    } else {
        // Set the state
        temp = pref + '0';

        // Goto next state
        solve(temp, n, ans);
    }

}

vector<string> generateString(int n) {
    vector<string> ans;
    string pref = "";
    solve(pref, n, ans);
    sort(ans.begin(), ans.end())

    return ans;

}

File: CodeStudio/binary-strings-with-no-consecutive-1s/main.py
from typing import List


def generateString(n: int) -> List[str]:

    def solve(pref, n, acc):
        if (len(pref) == n):
            acc.append(pref)
            return acc

        if (len(pref) < n):

            if (pref == '' or pref[-1] == '0'):

                # Set the state
                temp = pref + '0'

                # Goto next state
                solve(temp, n, acc)

                # Set the state
                temp = pref + '1'

                # Goto next state
                solve(temp, n, acc)

            else:

                # Set the state
                temp = pref + '0'

                # Goto next state
                solve(temp, n, acc)

            return acc

        return acc

    ans = solve('', n, [])

    # print('ans : ', ans)

    return ans
